lib/index.js

```diff
@@ -74,6 +79,21 @@ module.exports = function(playlistURL, options) {
       totalItems++;
       requestQueue.push(item, onQueuedEnd);
     });
+    parser.on('ytdl-end', function() {
```

Comment:
We do not want to call the normal end, as that sets a 10 minute timeout, when we need to call every 5 seconds. So, this custom function will do just that. It deletes the parser just in case that causes a memory leak (because we create a new one every loop and don't "return"), and then calls refreshPlaylist again. It finally sets ytft to false and rpts to 0 (see above).